cmd/gateway/s3/gateway-s3.go

```diff
@@ -635,6 +648,48 @@ func (l *s3Objects) DeleteBucketPolicy(ctx context.Context, bucket string) error
 	return nil
 }
 
+// GetObjectTag gets the tags set on the object
+func (l *s3Objects) GetObjectTag(ctx context.Context, bucket string, object string) (*tags.Tags, error) {
+	var err error
+	var tagObj *tags.Tags
+	var tagStr string
+	var opts minio.ObjectOptions
+
+	if _, err = l.GetObjectInfo(ctx, bucket, object, opts); err != nil {
+		return nil, minio.ErrorRespToObjectError(err, bucket, object)
+	}
+
+	if tagStr, err = l.Client.GetObjectTagging(bucket, object); err != nil {
+		return nil, minio.ErrorRespToObjectError(err, bucket, object)
+	}
+
+	if tagObj, err = tags.ParseObjectXML(strings.NewReader(tagStr)); err != nil {
+		return nil, minio.ErrorRespToObjectError(err, bucket, object)
+	}
+	return tagObj, err
+}
+
+//PutObjectTag attaches the tags to the object
+func (l *s3Objects) PutObjectTag(ctx context.Context, bucket, object string, tagStr string) error {
+	var err error
+	tagObj, err := tags.Parse(tagStr, true)
+	if err != nil {
+		return minio.ErrorRespToObjectError(err, bucket, object)
+	}
+	tagMap := tagObj.ToMap()
+	if err = l.Client.PutObjectTagging(bucket, object, tagMap); err != nil {
+		return minio.ErrorRespToObjectError(err, bucket, object)
+	}
+	return nil
+}
+
+func (l *s3Objects) DeleteObjectTag(ctx context.Context, bucket, object string) error {
```

Comment:
If `{Get, Put, Delete}ObjectTag()` are introduced in this PR, you could have plural `{Get, Put, Delete}ObjectTags()` or `{Get, Put, Delete}ObjectTagging()`

Reply:
I made this change. PTAL.